packages/xrpl/src/Wallet/fundWallet.ts

```diff
@@ -83,205 +91,134 @@ const MAX_ATTEMPTS = 20
+  })
 
+  const body = (await response.json()) as FaucetWallet
   // "application/json; charset=utf-8"
-  if (response.headers['content-type']?.startsWith('application/json')) {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- We know this is safe and correct
-    const faucetWallet: FaucetWallet = JSON.parse(body)
-    const classicAddress = faucetWallet.account.classicAddress
-    await processSuccessfulResponse(
+  if (
+    response.ok &&
+    response.headers.get('Content-Type')?.startsWith('application/json')
+  ) {
+    const classicAddress = body.account.classicAddress
+    return processSuccessfulResponse(
       client,
       classicAddress,
       walletToFund,
       startingBalance,
-      resolve,
-      reject,
-    )
-  } else {
-    reject(
-      new XRPLFaucetError(
-        `Content type is not \`application/json\`: ${JSON.stringify({
-          statusCode: response.statusCode,
-          contentType: response.headers['content-type'],
-          body,
-        })}`,
-      ),
     )
   }
+  return Promise.reject(
+    new XRPLFaucetError(
+      `Content type is not \`application/json\`: ${JSON.stringify({
+        body,
+        contentType: response.headers.get('Content-Type'),
+        statusCode: response.status,
+      })}`,
+    ),
+  )
+
+  // POST the body
 }
 
-// eslint-disable-next-line max-params, max-lines-per-function -- Only used as a helper function, lines inc due to added balance.
+// eslint-disable-next-line max-params -- Only used as a helper function, lines inc due to added balance.
 async function processSuccessfulResponse(
   client: Client,
   classicAddress: string | undefined,
   walletToFund: Wallet,
   startingBalance: number,
-  resolve: (response: { wallet: Wallet; balance: number }) => void,
-  reject: (err: ErrorConstructor | Error | unknown) => void,
-): Promise<void> {
+): Promise<{
+  wallet: Wallet
+  balance: number
+}> {
   if (!classicAddress) {
-    reject(new XRPLFaucetError(`The faucet account is undefined`))
-    return
+    return Promise.reject(
+      new XRPLFaucetError(`The faucet account is undefined`),
+    )
   }
   try {
     // Check at regular interval if the address is enabled on the XRPL and funded
-    const updatedBalance = await getUpdatedBalance(
+    const balance = await getUpdatedBalance(
       client,
       classicAddress,
       startingBalance,
     )
 
-    if (updatedBalance > startingBalance) {
-      resolve({
-        wallet: walletToFund,
-        balance: await getUpdatedBalance(
-          client,
-          walletToFund.classicAddress,
-          startingBalance,
-        ),
-      })
-    } else {
-      reject(
-        new XRPLFaucetError(
-          `Unable to fund address with faucet after waiting ${
-            INTERVAL_SECONDS * MAX_ATTEMPTS
-          } seconds`,
-        ),
-      )
+    return {
+      wallet: walletToFund,
+      balance,
     }
+
+    throw new XRPLFaucetError(
```

Comment:
did you get rid of comparing existing and updated balance logic?

Reply:
There was a catch

Reply:
It will always be greater than starting amount and then it would retrieve the balance yet again.

Reply:
oh yeah the logic should be if it isn't greater then retrieve again, instead of it is greater. There could be an edge case where the balance remained unchanged but the error in the transaction wasn't caught. if that never happens, `startingBalance` shouldn't be a param since it's not used?

Reply:
I just realized that I still needed the check just not the additional balance retrieval.